### PR TITLE
Added support for sending ServiceChecks

### PR DESCRIPTION
**Pull request by nathanrobb**
Implemented before reading issues (oops).

This should solve issue #23 Service checks?

### CODE REVIEW

**Comment by yannmh** [comment]
This looks great, thanks @nathanrobb 🙇 

I am adding your changes to our `2.2.0` milestone: it'll be reviewed shortly.

File: src/StatsdClient/Statsd.cs [hunk 1]
```diff
@@ -50,7 +50,22 @@ public abstract class Metric : ICommandType
...
+                    .Replace("\r", "")
+                    .Replace("\n", "\\n");
+            }
+
+            protected static string TruncateOverage(string str, int overage)
+            {
+                return str.Substring(0, str.Length - overage);
+            }
+        }
+
```
Comment (yannmh) [inline src/StatsdClient/Statsd.cs]: I am wondering if having these two methods defined `private` in the `Statsd` class is not enough. In particular, I found counter-intuitive that `Event`, `ServiceCheck` inherit from `Command` but not `Metric`. What do you think ?

File: src/Tests/MetricsAndEventsIntegrationTests.cs [hunk 2]
```diff
@@ -591,6 +591,20 @@ public void events_aggregation_key_and_tags()
...
+        }
+
+        [Test]
+        public void service_check_tags_message()
+        {
+            DogStatsd.ServiceCheck("na\r\nme", Status.CRITICAL, tags: new[] { "t1", "t2:v2" }, message: "m:mess\r\nage");
+            AssertWasReceived("_sc|na\\nme|2|#t1,t2:v2|m:m\\:mess\\nage");
+        }
     }
 }
```
Comment (yannmh) [inline src/Tests/MetricsAndEventsIntegrationTests.cs]: I guess it's time to rename `MetricsAndEventsIntegrationTests.cs` 🎉 . Maybe something like `CommandTests.cs` ?

File: src/StatsdClient/Statsd.cs [hunk 3]
```diff
@@ -151,10 +218,28 @@ public void Add(string title, string text, string alertType = null, string aggre
...
         }
+
         public void Send(string title, string text, string alertType = null, string aggregationKey = null, string sourceType = null, int? dateHappened = null, string priority = null, string hostname = null, string[] tags = null, bool truncateIfTooLong = false)
         {
             Send(Event.GetCommand(title, text, alertType, aggregationKey, sourceType, dateHappened, priority, hostname, tags, truncateIfTooLong));
         }
+
+        /// <summary>
+        /// Add a Service check
+        /// </summary>
```
Comment (yannmh) [inline src/StatsdClient/Statsd.cs]: 🍰

File: src/Tests/StatsdUnitTests.cs [hunk 4]
```diff
@@ -870,5 +870,238 @@ public void add_set_string_with_sample_rate_and_tags()
...
+        {
+            Statsd s = new Statsd(udp, _randomGenerator, _stopwatch);
+
+            var length = 8 * 1024 - 6;
+            var builder = BuildLongString(length);
+            var name = builder;
+
+            var exception = Assert.Throws<Exception>(() => s.Add(name + "x", 0));
+            Assert.That(exception.Message, Contains.Substring("payload is too big"));
+        }
```
Comment (yannmh) [inline src/Tests/StatsdUnitTests.cs]: Thanks for the great test coverage 🙇 .

**Comment by yannmh** [comment]
Great addition!
This is ready to be merged: I just added two nitpick, please let me know what you think about it. 

Thanks again @nathanrobb

File: src/Tests/MetricsAndEventsIntegrationTests.cs (hunk 2 above)
Comment (nathanrobb) [inline src/Tests/MetricsAndEventsIntegrationTests.cs]: Should be resolved in 16ea524.
Cheers.

File: src/StatsdClient/Statsd.cs (hunk 1 above)
Comment (nathanrobb) [inline src/StatsdClient/Statsd.cs]: I think I might be missing something here, sorry.

What are you meaning by this comment?

Are you saying it is odd that Metric doesn't inherit the new Command class?
I did feel this was a little yuck, those methods could be placed anywhere else to make it cleaner.

I'll hold off on any changes until I can get a little clarification.

Cheers.

Comment (yannmh) [inline src/StatsdClient/Statsd.cs]: > Are you saying it is odd that Metric doesn't inherit the new Command class?
Yes.  :)

> I did feel this was a little yuck, those methods could be placed anywhere else to make it cleaner.
I wonder if we could simply move these methods to `StatsD` class instead.

Comment (nathanrobb) [inline src/StatsdClient/Statsd.cs]: Sweet as, this has been fixed in 503f804.
Cheers.

**Comment by yannmh** [comment]
Thanks for your responsiveness @nathanrobb . 

Let's merge it: it'll come out with the `2.2.0` release 🚢 .

**Comment by nathanrobb** [comment]
Awesome, thanks for your help on this @yannmh

**Pull request merged**